ddl/foreign_key.go

```diff
@@ -269,3 +280,61 @@ func checkTableForeignKey(referTblInfo, tblInfo *model.TableInfo, fkInfo *model.
 	}
 	return nil
 }
+
+func checkTableHasForeignKeyReferred(is infoschema.InfoSchema, schema, tbl string, ignoreTables []ast.Ident, fkCheck bool) *model.ReferredFKInfo {
+	if !fkCheck {
+		return nil
+	}
+	referredFKs := is.GetTableReferredForeignKeys(schema, tbl)
+	for _, referredFK := range referredFKs {
+		found := false
+		for _, tb := range ignoreTables {
+			if referredFK.ChildSchema.L == tb.Schema.L && referredFK.ChildTable.L == tb.Name.L {
+				found = true
+				break
+			}
+		}
+		if found {
+			continue
+		}
+		if is.TableExists(referredFK.ChildSchema, referredFK.ChildTable) {
+			return referredFK
+		}
+	}
+	return nil
+}
+
+func checkDropTableHasForeignKeyReferredInOwner(d *ddlCtx, t *meta.Meta, job *model.Job) error {
+	if !variable.EnableForeignKey.Load() {
+		return nil
+	}
+	var objectIdents []ast.Ident
+	var fkCheck bool
+	err := job.DecodeArgs(&objectIdents, &fkCheck)
+	if err != nil {
+		job.State = model.JobStateCancelled
+		return errors.Trace(err)
+	}
+	referredFK, err := checkTableHasForeignKeyReferredInOwner(d, t, job.SchemaName, job.TableName, objectIdents, fkCheck)
```

Comment:
Dropping tables is not atomic. If only parent table is dropped successfully but child table failed, is it acceptable?

Reply:
I already move forward this check. 
Besides, if the parent table is dropped but the child table still there, it is acceptable, since if the user `set @@foreign_key_check=0`, then the user can drop the parent table anyway.